### PR TITLE
interface of dynamic reader block filter for order by.

### PR DESCRIPTION
**Pull request by gouhongshen**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #https://github.com/matrixorigin/matrixone/issues/11805

## What this PR does / why we need it:
make block reader simple.

### CODE REVIEW

File: pkg/sql/colexec/table_scan/table_scan.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ func (tableScan *TableScan) OpType() vm.OpType {
 
 func (tableScan *TableScan) Prepare(proc *process.Process) (err error) {
 	tableScan.ctr = new(container)
-	tableScan.ctr.orderBy = tableScan.Reader.GetOrderBy()
+	//tableScan.ctr.orderBy = tableScan.Reader.GetOrderBy()
```
Comment (fengttt) [inline pkg/sql/colexec/table_scan/table_scan.go:41]: Why?

File: pkg/sql/colexec/table_scan/table_scan.go [hunk 2]
```diff
@@ -101,12 +101,12 @@ func (tableScan *TableScan) Call(proc *process.Process) (vm.CallResult, error) {
 		// receive topvalue message
 		if tableScan.ctr.msgReceiver != nil {
 			msgs, _ := tableScan.ctr.msgReceiver.ReceiveMessage(false, proc.Ctx)
-			for i := range msgs {
-				msg, ok := msgs[i].(process.TopValueMessage)
-				if !ok {
-					panic("only support top value message in table scan!")
-				}
-				tableScan.Reader.SetFilterZM(msg.TopValueZM)
+			for _ = range msgs {
```
Comment (fengttt) [inline pkg/sql/colexec/table_scan/table_scan.go:104]: Here too.   Why do you comment it out?   Either add enough doc/comment on why, or if the code is dead, just remove.

File: pkg/sql/compile/scope.go [hunk 3]
```diff
@@ -563,7 +563,7 @@ func buildScanParallelRun(s *Scope, c *Compile) (*Scope, error) {
 	if scanUsedCpuNumber == 1 {
 		s.Magic = Normal
 		s.DataSource.R = readers[0]
-		s.DataSource.R.SetOrderBy(s.DataSource.OrderBy)
+		//s.DataSource.R.SetOrderBy(s.DataSource.OrderBy)
```
Comment (fengttt) [inline pkg/sql/compile/scope.go:566]: Here tool.

**Review by fengttt — CHANGES_REQUESTED** [review]
Please fix commented code.

**Review by daviszhen — APPROVED** [review]

**Pull request closed without merging**